packages/medusa/src/services/swap.ts

```diff
@@ -802,7 +796,13 @@ class SwapService extends TransactionBaseService {
 
       swap.difference_due = total
       swap.shipping_address_id = cart.shipping_address_id
-      swap.shipping_methods = cart.shipping_methods
+      // TODO: Due to cascade insert we have to remove the tax_lines that have been added by the cart decorate totals.
+      // Is the cascade insert really used? Also, is it really necessary to pass the entire entities when creating or updating?
```

Comment:
comment: I think it would be appropriate to remove the cascade on shipping method in the swap model, but can be left for a follow-up iteration 👍

Reply:
Yeah it is what I thought, I keep the todo for later refactoring to not make the pr bigger